### PR TITLE
[COM-1066] Allow overriding any key/value in a view or event

### PR DESCRIPTION
**Pull request by francoislg**
This allows sending custom data with the `event` and `pageview` types.

Use cases for that include:

* Overriding an existing key for one call
* Adding the `custom` key with custom values to be forwarded in the collect endpoint
* ... and much more! (I think)

I added a test to validate that it works, and changed a bit the `setup` methods of the tests to improve the feedback a bit.

I also removed the `Purpose` column since it's more confusing than not with the collect endpoint. I allowed me to remove some code

### CODE REVIEW

**Comment by toussaints** [comment]
I'm pretty sure you haven't used the right JIRA #
Probably falls down under COM-1066

**Comment by francoislg** [comment]
Indeed, no idea how I messed it up :P

File: template.tpl [hunk 1]
```diff
@@ -227,11 +227,21 @@ ___TEMPLATE_PARAMETERS___
...
+      {
+        "paramName": "eventType",
+        "paramValue": "event",
+        "type": "EQUALS"
       }
     ],
     "displayName": "Custom metadata (leverageable by Coveo ML and in Coveo UA reports)",
     "name": "Custom Data",
-    "groupStyle": "ZIPPY_OPEN",
+    "groupStyle": "ZIPPY_CLOSED",
```
Comment (toussaints) [inline template.tpl:244]: image

File: template.tpl [hunk 2]
```diff
@@ -259,26 +269,6 @@ ___TEMPLATE_PARAMETERS___
...
-                "displayValue": "Coveo ML context",
-                "value": "ml"
-              },
-              {
-                "displayValue": "All",
-                "value": "all"
-              }
-            ],
-            "defaultValue": "usageanalytics",
-            "displayName": "Purpose",
```
Comment (toussaints) [inline template.tpl:279]: Is it breaking when upgrading the library?

File: template.tpl [hunk 3]
```diff
@@ -1001,6 +970,16 @@ scenarios:
     \ = {\n  \"eventType\": \"view\"\n};\n\nrunCode(mockData);\n\nassertApi('gtmOnFailure').wasNotCalled();\n\
     assertApi('gtmOnSuccess').wasCalled();\nassertActionWasQueued([\"set\", \"currencyCode\"\
     , \"EUR\"]);\nassertEventWasSentWith(\"pageview\");"
+- name: Adds Impressions with a addToCart event
+  code: "const product = {\n   \"productName\": \"wow\" \n};\n\nconst impression =\
```
Comment (toussaints) [inline template.tpl:974]: those are so cryptic to read

File: template.tpl [hunk 4]
```diff
@@ -639,6 +602,9 @@ ___WEB_PERMISSIONS___
         }
       ]
     },
+    "clientAnnotations": {
+      "isEditedByUser": true
```
Comment (toussaints) [inline template.tpl:606]: not sure what this does

File: template.tpl [hunk 5]
```diff
@@ -1010,9 +989,40 @@ scenarios:
     assertApi('gtmOnSuccess').wasCalled();\nassertActionWasQueued([\"ec:setAction\"\
     , \"impression\", { \"list\": \"coveo:search:1234\" }]);\nassertActionWasQueued([\"\
     ec:addImpression\", product]);\nassertEventWasSentWith(\"event\");"
+
+- name: Sends a Page View event with custom data
```
Comment (toussaints) [inline template.tpl:993]: why is this one well formatted while the others aren't?

**Review by toussaints — APPROVED** [review]
LGTM

File: template.tpl (hunk 2 above)
Comment (francoislg) [reply template.tpl:279]: :thinking: the field should just disapear

File: template.tpl (hunk 4 above)
Comment (francoislg) [reply template.tpl:606]: I don't know either, it appear on the export so I left them in

File: template.tpl (hunk 5 above)
Comment (francoislg) [reply template.tpl:993]: Huh, they probably changed something for the new tests

**Review by dlafreniere — APPROVED** [review]

**Pull request merged**